src/query/shorthand.ts

```diff
@@ -429,9 +430,9 @@ export namespace shorthandParser {
     return encQ;
   }
 
-  export function rawFieldDef(encQ: EncodingQuery, fieldDefPart: string[]): EncodingQuery {
-    encQ.field = fieldDefPart[0];
-    encQ.type = getFullName(fieldDefPart[1].toUpperCase()) || '?';
+  export function rawFieldDef(fieldQ: FieldQuery, fieldDefPart: string[]): FieldQuery {
```

Comment:
We should rename this to rawEncodingDef and take `encQ: EncodingQuery` as input. 

Depending on `fieldDefPart[0]`, we can assign the value to `field` or `value`. 

If you follow [this proposal](https://github.com/vega/compassql/pull/335/files#r101200446), then you probably can use the following rules:

If it can be parsed by JSON.parse, assign to `value`.
If it's an unquoted string, assign to `field`.